data/cube/View.js

```diff
@@ -94,8 +120,16 @@ export class View extends HoistBase {
     @action
     updateQuery(overrides) {
         throwIf(overrides.cube, 'Cannot redirect view to a different cube in updateQuery().');
-        this.query = this.query.clone(overrides);
-        this.fullUpdate();
+        const newQuery = this.query.clone(overrides);
+        if (this.query.equals(newQuery)) return;
+
+        // If anything other than just filter changing, blow away row cache
+        if (!isEqual(keys(overrides), ['filter'])) {
+            this._rowCache.clear();
+        }
+
+        this.query = newQuery;
+        this.fullUpdateAsync().linkTo(this.loadModel);
```

Comment:
I think we need to await this and make updateQuery async - having trouble with my masking operations in the client app

Reply:
am looking at this now.  Not sure why this method needs to block on the update being completed?  Seems like potentially a different problem.  Are you passing in a shared PendingTaskModel?  Is it set mode = 'all'?

Reply:
ah! i think that's it! it's set to the 'last' default mode